File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cs
using System;

public class Solution {
    public int solution(int[,] sizes) {
        int answer = 0;
        int maxW = 0; int maxH = 0;

        for(int i = 0; i < sizes.GetLength(0); i++)
        {
            int big = Math.Max(sizes[i, 0], sizes[i, 1]);
            int small = Math.Min(sizes[i, 0], sizes[i, 1]);
            maxW = Math.Max(big, maxW);
            maxH = Math.Max(small, maxH);
        }
        answer = maxW * maxH;
        return answer;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int maxW = 0;
    int maxH = 0;
    for(auto s : sizes)
    {
        int w = max(s[0], s[1]);
        int h = min(s[0], s[1]);
        
        maxW = max(maxW, w);
        maxH = max(maxH, h);
    }
    return maxW * maxH;
}

